onnx/defs/tensor/defs.cc

```diff
@@ -1595,6 +1664,28 @@ ONNX_OPERATOR_SET_SCHEMA(
                                             : // i - axis < q
                     data_shape.dim(i - q + 1); // i < out_rank < q + r - 1
           }
+        })
+        .PartialDataPropagationFunction([](DataPropagationContext& ctx) {
+          if (!axisIsZero(ctx, true)) {
```

Comment:
Actually, I wonder if we should restrict ourselves to the case where input(0) is 1-dimensional tensor? If it is multidimensional, then it is more complicated.

Reply:
agree

Reply:
We do have this restriction now for every data propagation, right? (only 0D or 1D tensor can be getInputData)
For other N-D tensors (which N > 1), their getInputData() will be nullptr and it will stop data propagation.